addons/im_livechat/static/src/embed/common/chat_window_model_patch.js

```diff
@@ -16,6 +16,8 @@ patch(ChatWindow.prototype, {
             this.open({ notifyState: this.thread?.state !== "open" });
         } else {
             await super.close(...arguments);
+            this.thread.messages.forEach((message) => message.delete());
+            this.thread.delete();
```

Comment:
I would move to models, just because async code could not be executed when the component is destroyed, which is likely here because `.close()` deletes chat window model thus chat window component should be deleted too